### PR TITLE
support display of labels in autocompletion popup

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/14242.

Screenshot 2024-02-08 at 11 13 18 AM

### Approach

Some R variables might be annotated with a "label" attribute. This PR bundles that information as part of the completion metadata when available, and uses that in the displayed completion list.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/14242.

### Documentation

Not included; not sure if this is worth documenting.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Comment by kevinushey** [comment]
I wonder if I should alter the styling for the metadata included, just to ensure it's distinct from the completion itself?

File: src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/assist/CompletionRequester.java [hunk 1]
```diff
@@ -312,6 +313,13 @@ public void onError(ServerError error)
          {
             callback.onError(error);
          }
+         
+         private boolean isPriorityCompletion(int type)
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/assist/CompletionRequester.java:317]: This helps fix an issue where column-based completions were displayed below so-called "context" completions in `[` and `[[` completion contexts, e.g.

Screenshot 2024-02-08 at 11 19 13 AM

**Review by gtritchie — APPROVED** [review]
LGTM!

**Comment by kevinushey** [comment]
Now also filtering on the whole display string:

Screenshot 2024-02-08 at 1 22 46 PM

**Pull request merged**